src/Validators/OptionalIfCheck.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Validators.IntegrityChecks.Form
+{
+    public class OptionalIfCheck : IFormSchemaIntegrityCheck
+    {
+        public IntegrityCheckResult Validate(FormSchema schema)
+        {
+            IntegrityCheckResult result = new();
+            List<string> questionIds = new();
+
+            List<IElement> elementsWithOptionalIfInJSON = schema.Pages
+                .Where(page => page.Elements is not null)
+                .SelectMany(page => page.Elements)
```

Comment:
Maybe switch to the ValidatableElements as that's what your main logic is based off now...